lib/scalingo/api/response.rb

```diff
@@ -5,7 +5,7 @@ def self.unpack(client, response, key: nil)
         body = response.body
         has_hash_body = body.present? && body.respond_to?(:key)
 
-        data = has_hash_body && key.present? ? body[key] : body
+        data = has_hash_body && key.present? && response.success? ? body[key] : body
```

Comment:
```suggestion
        data = if response.success? && has_hash_body && key.present?
            body[key] 
          else
            body
          end
```

I find it kind of hard to read. What do you think?

Reply:
Yeah ok, you have a point. I rewrote it slightly differently :)